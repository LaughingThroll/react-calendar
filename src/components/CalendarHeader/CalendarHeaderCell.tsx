import React from 'react'
import classnames from 'classnames'

import { isWeekend, getDayInBinaryString } from '../../utils/date/index'

interface ICalendarHeaderCell {
  date: Date
}

const CalendarHeaderCell: React.FC<ICalendarHeaderCell> = ({ date }) => {
  return (
    <th
      className={classnames({
        'calendar-header__cell': true,
        'cell-gray': isWeekend(date),
      })}
    >
      <span className="calendar-header__day">{getDayInBinaryString(date)}</span>
      <span className="calendar-header__number">{date.getDate()}</span>
    </th>
  )
}

export default React.memo(CalendarHeaderCell)
