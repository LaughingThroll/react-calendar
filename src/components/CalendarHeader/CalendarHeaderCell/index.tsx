import React from 'react'

import Cell from '../../common/Cells/Cell'

import { getDayInBinaryString } from '../../../utils/date'

import { CellVariants } from '../../../types/cell'

export interface CalendarHeaderCellProps {
  cellDate: Date
  classNames?: string[]
}

const CalendarHeaderCell: React.FC<CalendarHeaderCellProps> = ({ cellDate, classNames = [] }) => {
  return (
    <Cell cellDate={cellDate} variant={CellVariants.TH} classNames={classNames}>
      <span className="calendar-header__day">{getDayInBinaryString(cellDate)}</span>
      <span className="calendar-header__number">{cellDate.getDate()}</span>
    </Cell>
  )
}

export default React.memo(CalendarHeaderCell)
