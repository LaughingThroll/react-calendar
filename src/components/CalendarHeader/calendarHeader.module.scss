@import 'styles/utils/mixin';

.calendarHeader {
  &__cell {
    &:first-child {
      border-top: 0;
      border-left: 0;
      padding-bottom: 24px;
      width: 260px;
    }

    &:not(:first-child) {
      & span:first-child {
        font-weight: 400;
      }

      & > * {
        display: block;
      }
    }

    &:last-child {
      font-weight: 400;
    }
  }

  &__day {
    font-weight: 400;
  }

  &__number {
    @extend %font-bold;
  }
}
