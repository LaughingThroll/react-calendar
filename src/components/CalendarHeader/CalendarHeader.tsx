import React from "react"

import Button from "../common/Button"
import CalendarHeaderCell from "./CalendarHeaderCell"

import { formatDayInBinaryString } from "../../utils/date"

interface ICalendarHeader {
  date: Date
  daysInMonth: number
}

const CalendarHeader: React.FC<ICalendarHeader> = ({ date, daysInMonth }) => {
  return (
    <thead>
      <tr className="calendar-header">
        <th className="calendar-header__cell">
          <Button iconPlus>Add Vacation</Button>
        </th>
        {new Array(daysInMonth).fill(0).map((_, day) => (
          <CalendarHeaderCell key={day} dayString={formatDayInBinaryString(date, day + 1)} dayNumber={day + 1} />
        ))}
        <th className="calendar-header__cell cell-gray"> Sum </th>
      </tr>
    </thead>
  )
}

export default CalendarHeader
