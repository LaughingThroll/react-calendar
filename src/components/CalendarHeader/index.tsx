import React from 'react'

import styles from './calendarHeader.module.scss'

import CalendarHeaderCell from './CalendarHeaderCell'
import { CellSumm } from '../common'

import { getAllDaysInMonth } from '../../utils/date'

export interface CalendarHeaderProps {
  date: Date
  children?: any
  classNames?: string[]
}

const { calendarHeader, calendarHeader__row, calendarHeader__cell } = styles

const CalendarHeader: React.FC<CalendarHeaderProps> = ({ date, children, classNames = [] }) => {
  return (
    <thead className={[calendarHeader, ...classNames].join(' ')}>
      <tr className={calendarHeader__row}>
        <th className={calendarHeader__cell}>{children}</th>
        {getAllDaysInMonth(date).map((date, index) => (
          <CalendarHeaderCell key={index} cellDate={date} classNames={[calendarHeader__cell]} />
        ))}
        <CellSumm classNames={[calendarHeader__cell]}>Sum </CellSumm>
      </tr>
    </thead>
  )
}

export default React.memo(CalendarHeader)
