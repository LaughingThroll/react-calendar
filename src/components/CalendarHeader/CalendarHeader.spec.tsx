import { shallow } from 'enzyme'
import CalendarHeader, { CalendarHeaderProps } from './'

const setUp = (props: CalendarHeaderProps) => shallow(<CalendarHeader {...props} />)

describe('CalendarHeader render', () => {
  it('default render', () => {
    const props = { date: new Date('2021-05-05') }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
  })
})
