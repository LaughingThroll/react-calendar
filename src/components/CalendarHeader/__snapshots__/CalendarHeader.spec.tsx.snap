// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarHeader render default render 1`] = `
<thead
  className="calendarHeader"
>
  <tr
    className="calendarHeader__row"
  >
    <th
      className="calendarHeader__cell"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-04-30T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="0"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-01T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="1"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-02T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="2"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-03T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="3"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-04T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="4"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-05T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="5"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-06T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="6"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-07T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="7"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-08T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="8"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-09T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="9"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-10T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="10"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-11T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="11"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-12T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="12"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-13T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="13"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-14T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="14"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-15T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="15"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-16T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="16"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-17T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="17"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-18T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="18"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-19T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="19"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-20T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="20"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-21T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="21"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-22T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="22"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-23T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="23"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-24T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="24"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-25T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="25"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-26T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="26"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-27T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="27"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-28T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="28"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-29T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="29"
    />
    <Memo(CalendarHeaderCell)
      cellDate={2021-05-30T21:00:00.000Z}
      classNames={
        Array [
          "calendarHeader__cell",
        ]
      }
      key="30"
    />
    <Memo(CellSumm)
      classNames={
        Array [
          "calendarHeader__cell",
          "cell-gray",
        ]
      }
    >
      Sum 
    </Memo(CellSumm)>
  </tr>
</thead>
`;
